animal_test.go

```diff
@@ -39,20 +39,20 @@ func TestAskUserYesOrNo(t *testing.T) {
 
 func TestGetUserYesOrNo(t *testing.T) {
 	t.Parallel()
-	var stdin bytes.Buffer
-	stdin.Write([]byte("yes\n"))
+	var input bytes.Buffer
+	input.Write([]byte("yes"))
 	testCases := []struct {
 		question string
 		want     string
 	}{
 		{question: "Is it a horse", want: "yes"},
```

Comment:
Once you have this working, we might consider additional test cases. For example, we might want to accept `"yes"`, `"y"`, `"Yes"`, or `"YES"` as indicating yes. So we could add a field `input` to the test case struct with some of these variants, and put that string into the buffer before calling the function. The same goes for 'no' answers.

Reply:
Added this during our mentoring session.